src/providers/v2/quote-provider.ts

```diff
@@ -92,62 +90,10 @@ export class V2QuoteProvider implements IV2QuoteProvider {
             let outputAmount = amount.wrapped;
 
             for (const pair of route.pairs) {
-              if (amount.wrapped.currency.sellFeeBps) {
-                // this should never happen, but just in case it happens,
-                // there is a bug in sor. We need to log this and investigate.
-                const error =
-                  new Error(`Sell fee bps should not exist on output amount
-                ${JSON.stringify(amount)} on amounts ${JSON.stringify(amounts)}
-                on routes ${JSON.stringify(routes)}`);
-
-                // artificially create error object and pass in log.error so that
-                // it also log the stack trace
-                log.error(
-                  { error },
-                  'Sell fee bps should not exist on output amount'
-                );
-                metric.putMetric(
-                  'V2_QUOTE_PROVIDER_INCONSISTENT_SELL_FEE_BPS_VS_FEATURE_FLAG',
-                  1,
-                  MetricLoggerUnit.Count
-                );
-              }
-
-              if (providerConfig.enableFeeOnTransferFeeFetching) {
-                if (
-                  pair.token0.equals(outputAmount.currency) &&
-                  pair.token0.sellFeeBps?.gt(BigNumber.from(0))
-                ) {
-                  const outputAmountWithSellFeeBps =
-                    CurrencyAmount.fromRawAmount(
-                      pair.token0,
-                      outputAmount.quotient
-                    );
-                  const [outputAmountNew] = pair.getOutputAmount(
-                    outputAmountWithSellFeeBps
-                  );
-                  outputAmount = outputAmountNew;
-                } else if (
-                  pair.token1.equals(outputAmount.currency) &&
-                  pair.token1.sellFeeBps?.gt(BigNumber.from(0))
-                ) {
-                  const outputAmountWithSellFeeBps =
-                    CurrencyAmount.fromRawAmount(
-                      pair.token1,
-                      outputAmount.quotient
-                    );
-                  const [outputAmountNew] = pair.getOutputAmount(
-                    outputAmountWithSellFeeBps
-                  );
-                  outputAmount = outputAmountNew;
-                } else {
-                  const [outputAmountNew] = pair.getOutputAmount(outputAmount);
-                  outputAmount = outputAmountNew;
-                }
-              } else {
-                const [outputAmountNew] = pair.getOutputAmount(outputAmount);
-                outputAmount = outputAmountNew;
-              }
+              [outputAmount] = pair.getOutputAmount(
+                outputAmount,
+                providerConfig.enableFeeOnTransferFeeFetching ?? false
```

Comment:
nit - this reads a little weird. `providerConfig.enableFeeOnTransferFeeFetching === true` seems more direct.